colossalai/inference/server/api_server.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import json
+
+import uvicorn
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse, Response, StreamingResponse
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from colossalai.inference.config import InferenceConfig
+from colossalai.inference.core.engine import InferenceEngine
+from colossalai.inference.server.utils import id_generator
+
+TIMEOUT_KEEP_ALIVE = 5  # seconds.
+app = FastAPI()
+engine = None
+models_dict = {"Llama_Models": ("llama2-7b",)}
+
+
+@app.get("/v0/models")
+def get_available_models() -> Response:
+    return JSONResponse(models_dict)
+
+
+@app.post("/generate")
+async def generate(request: Request) -> Response:
+    """Generate completion for the request.
+
+    A request should be a JSON object with the following fields:
+    - prompts: the prompts to use for the generation.
+    - stream: whether to stream the results or not.
+    - other fields:
+    """
+    request_dict = await request.json()
+    prompt = request_dict.pop("prompt")
+    stream = request_dict.pop("stream", None)
+    request_id = id_generator()
+
+    results = engine.generate(
```

Comment:
It may be necessary to pass in the generation_config to configure the do_sample for this inference process.

Reply:
It's true and I will add it

Reply:
added